### ISSUE TITLE
cannot read property 'length' of null

### ISSUE THREAD

**Issue by kzidane** [cs50/harvard.cs50.droplet#56]
```
workspace-cs50.js:13499 Uncaught TypeError: Cannot read property 'length' of null
  at a.<anonymous> (https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:13499:48)
  at a.s.setPalette (https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:13500:11)
  at Array.<anonymous> (https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:22097:26)
  at EditSession.EventEmitter._emit.EventEmitter._dispatchEvent (https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:25372:46)
  at EditSession.$onChangeMode (https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:32022:6)
  at EditSession.<anonymous> (https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:32010:48)
  at https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:30045:78
  at Array.forEach (<anonymous>:null:null)
  at https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:30045:45
  at _require (https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:105:27)
  at https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:71:23
  at Array.forEach (<anonymous>:null:null)
  at processLoadQueue (https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:70:14)
  at HTMLScriptElement.s.onload.s.onreadystatechange (https://static.c9.io/nc-3.1.3681-06667a95/config/workspace-cs50.js:145:10)
```

The log from production might not help much, but it might give you a clue of what's wrong. The way I've been able to replicate is just creating a new file, File > Save As ... > name the file > Save.

![saveas](https://user-images.githubusercontent.com/7230211/28325190-2b35f9f0-6bab-11e7-98b5-ba549ead62f1.gif)

I'll try to test offline later, see if I can get a more useful log.

**Comment by dlloyd09** [comment]
Seems to have issues for empty files. blocks mode shouldn't try to convert if file length == 0 or if non-C/C++.

**Comment by dabbler0** [comment]
I figured out the cause of this bug, which happens whenever Droplet switches into a non-C++ mode. It was fixed, however, in commit 2602787ff3c08a92eb137b118960f7e982158e40, on line 598. It doesn't occur in current master. Closing now.

**Comment by dmalan** [comment]
Awesome, thanks.
